src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs

```diff
@@ -81,6 +87,12 @@ internal static string GetParameterText(string prefix, string parameterName, ILi
                     {
                         sb.Append("\n");
                     }
+
+                    if (!Traits.Instance.EscapeHatches.LogTaskInputsVerbose && sb.Length >= parameterTextLimit)
+                    {
+                        sb.Append("...\nThe parameters have been truncated beyond this point. To view all parameters, set environment variable MSBUILDLOGTASKINPUTSVERBOSE=1");
```

Comment:
Use ResourceUtilities.FormatString (possibly with an option)

Reply:
To format what exactly?

Reply:
Here's an example: https://github.com/microsoft/msbuild/blob/master/src/Build/BackEnd/BuildManager/BuildManager.cs#L1983